inductiva/projects/project.py

```diff
@@ -0,0 +1,240 @@
+        """Updates itself with information fetched from the backend."""
+        self._info = ProjectInfo(**model.get("task_status_overview"))
+        self.created_at = model.get("created_at")
+        self.num_tasks = model.get("num_tasks")
+        self.name = model.get("name")
+        self.id = model.get("id")
+
+        return self
+
+    def open(self):
+        """Opens the project.
+
+        Open the project and make it the active one in the calling
+        thread. An opened project will ensure that calls to the
+        `get_current_project` function will return this project.
+        Consecutive calls to this method are idempotent.
+
+        Raises:
+            RuntimeError if another opened project exists, i.e.,
+            `get_current_project` returns something other than None.
+
+        """
+        _logger.debug("Opening project %s", self.name)
+        current_project = get_current_project()
+
+        if current_project is self:
+            _logger.debug("Project is already opened.")
+            return
+
+        if current_project is not None:
+            raise RuntimeError(
+                "Trying to open a project when another is active.")
+
+        self._token = _CURRENT_PROJECT.set(self)
+
+    def close(self):
+        """Closes the project.
+
+        Calls to the `get_current_project` will return `None` after
+        the project is closed.  Consecutive calls to this method are
+        idempotent.
+
+        """
+        if self._token is None:
+            return
+
+        _CURRENT_PROJECT.reset(self._token)
+        self._token = None
+
+    @property
+    def opened(self) -> bool:
+        """Checks if the project is open."""
+        return self._token is not None
+
+    @property
+    def info(self) -> ProjectInfo:
+        """Returns project info.
+
+        It makes no requests to the backend. Instead it uses
+        previously stored information. Therefore it can be
+        outdated. For a method that requests the most recent data from
+        the backend use: `get_info`.
+
+        """
+        return self._info
+
+    def get_info(self) -> ProjectInfo:
+        """Returns and updates project info.
+
+        Updates itself with info from the backend.
+
+        """
+        name = self.name
+        model = self._get_model(name)
+        self._update_from_api_response(model)
+        return self._info
+
+    def __str__(self) -> str:
+        return f"Project '{self.name}' with "\
+               f"{self.num_tasks} tasks (id={self.id})"
+
+    def desc(self) -> str:
+        header = str(self) + "\n"
+        summary = "\n".join(
+            f"  {k}: {v}" for k, v in self._info.task_by_status.items())
+        return header + summary
+
+    def get_tasks(self,
+                  last_n: int = 5,
+                  status: Optional[Union[str, models.TaskStatusCode]] = None):
+        """Fetches the tasks managed by this project from the backend."""
```

Comment:
missing argument description. Also, how do we specify that we want all tasks?

```suggestion
        """Get the last N submitted tasks to this project.
        
        Get the last N submitted tasks that belong to this project, eventually
        filtered by status. By default, only the last 5 submitted tasks are
        returned, irrespectively of their status.
        
        Args:
            last_n (int): ....
            status: ...
        """
```